tools/wheels/gfortran_utils.sh

```diff
@@ -0,0 +1,170 @@
+        *) echo Did not recognize plat $plat; return 1 ;;
+    esac
+    local uname=${2:-$(uname)}
+    if [ "$uname" != "Darwin" ]; then
+        if [ "$plat" == "intel" ]; then
+            echo plat=intel not allowed for Manylinux
+            return 1
+        fi
+        echo "manylinux1_$plat"
+        return
+    fi
+    # macOS 32-bit arch is i386
+    [ "$plat" == "i686" ] && plat="i386"
+    local target=$(echo $MACOSX_DEPLOYMENT_TARGET | tr .- _)
+    echo "macosx_${target}_${plat}"
+}
+
+function get_distutils_platform_ex {
+    # Report platform as in form of distutils get_platform.
+    # This is like the platform tag that pip will use.
+    # Modify fat architecture tags on macOS to reflect compiled architecture
+    # For non-darwin, report manylinux version
+    local plat=$1
+    local mb_ml_ver=${MB_ML_VER:-1}
+    case $plat in
+        i686|x86_64|arm64|universal2|intel|aarch64|s390x|ppc64le) ;;
+        *) echo Did not recognize plat $plat; return 1 ;;
+    esac
+    local uname=${2:-$(uname)}
+    if [ "$uname" != "Darwin" ]; then
+        if [ "$plat" == "intel" ]; then
+            echo plat=intel not allowed for Manylinux
+            return 1
+        fi
+        echo "manylinux${mb_ml_ver}_${plat}"
+        return
+    fi
+    # macOS 32-bit arch is i386
+    [ "$plat" == "i686" ] && plat="i386"
+    local target=$(echo $MACOSX_DEPLOYMENT_TARGET | tr .- _)
+    echo "macosx_${target}_${plat}"
+}
+
+function get_macosx_target {
+    # Report MACOSX_DEPLOYMENT_TARGET as given by distutils get_platform.
+    python -c "import sysconfig as s; print(s.get_config_vars()['MACOSX_DEPLOYMENT_TARGET'])"
+}
+
+function check_gfortran {
+    # Check that gfortran exists on the path
+    if [ -z "$(which gfortran)" ]; then
+        echo Missing gfortran
+        exit 1
+    fi
+}
+
+function get_gf_lib_for_suf {
+    local suffix=$1
+    local prefix=$2
+    local plat=${3:-$PLAT}
+    local uname=${4:-$(uname)}
+    if [ -z "$prefix" ]; then echo Prefix not defined; exit 1; fi
+    local plat_tag=$(get_distutils_platform_ex $plat $uname)
+    if [ -n "$suffix" ]; then suffix="-$suffix"; fi
+    local fname="$prefix-${plat_tag}${suffix}.tar.gz"
+    local out_fname="${ARCHIVE_SDIR}/$fname"
+    if [ ! -e "$out_fname" ]; then
+        curl -L "${GF_LIB_URL}/$fname" > $out_fname || (echo "Fetch of $out_fname failed"; exit 1)
+    fi
+    [ -s $out_fname ] || (echo "$out_fname is empty"; exit 24)
+    echo "$out_fname"
+}
+
+if [ "$(uname)" == "Darwin" ]; then
+    mac_target=${MACOSX_DEPLOYMENT_TARGET:-$(get_macosx_target)}
+    export MACOSX_DEPLOYMENT_TARGET=$mac_target
+    GFORTRAN_DMG="${GF_UTIL_DIR}/archives/gfortran-4.9.0-Mavericks.dmg"
+    export GFORTRAN_SHA="$(shasum $GFORTRAN_DMG)"
+
+    function install_arm64_cross_gfortran {
+        curl -L -O https://github.com/isuruf/gcc/releases/download/gcc-10-arm-20210228/gfortran-darwin-arm64.tar.gz
+        export GFORTRAN_SHA=f26990f6f08e19b2ec150b9da9d59bd0558261dd
```

Comment:
Try this

```suggestion
        curl -L -O https://github.com/isuruf/gcc/releases/download/gcc-10-arm-20210728/gfortran-darwin-arm64.tar.gz
        export GFORTRAN_SHA=4a1354e61294d5163609e83b6b2b082bd9a9bbdf
```

Reply:
I'll try that. Your previous suggestion allows a simple Fortran program to compile, but I then  see:
```
meson.build:54:0: ERROR: Unable to detect GNU compiler type:
  gfortran: fatal error: cannot execute ‘cc1’: execvp: No such file or directory
  compilation terminated.
```
when I try configuring the meson build.

Reply:
Fixed in that build

Reply:
Thanks so much, that fixed the gfortran issues, onto other roadblocks now.